modules/i3s/src/lib/helpers/i3s-nodepages-tiles.ts

```diff
@@ -10,11 +10,12 @@ import {getUrlWithToken, generateTilesetAttributeUrls} from '../utils/url-utils'
  */
 export default class I3SNodePagesTiles {
   tileset: Tileset;
-  nodePages: NodePage[];
+  nodePages: NodePage[] = [];
```

Comment:
Do we need this book keeping? Why not just change the type to Promise and keep the old code?

```suggestion
  nodePages: Promise<NodePage>[] = [];
```

Reply:
A node page contain 64 nodes. We need to keep the node page to be able to return node that wasn't requested before but another node from the same node page was. Otherwise the same node page will be requested 64 times.